buildSrc/src/main/java/org/elasticsearch/gradle/NoticeTask.java

```diff
@@ -0,0 +1,200 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+
+import java.io.IOException;
+import java.io.File;
+import java.io.FileReader;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.stream.Stream;
+
+/**
+ * A task to create a notice file which includes dependencies' notices.
+ */
+public class NoticeTask extends DefaultTask {
+
+    @InputFile
+    private File inputFile = getProject().getRootProject().file("NOTICE.txt");
+    @OutputFile
+    private File outputFile = new File(getProject().getBuildDir(), "notices/" + getName() + "/NOTICE.txt");
+
+    /** Directories to include notices from */
+    private List<File> licensesDirs = new ArrayList<>();
+
+    public File getInputFile() {
+        return inputFile;
+    }
+
+    public void setInputFile(File inputFile) {
+        this.inputFile = inputFile;
+    }
+
+    public File getOutputFile() {
+        return outputFile;
+    }
+
+    public void setOutputFile(File outputFile) {
+        this.outputFile = outputFile;
+    }
+
+    public void licensesDir(File licensesDir) {
+        licensesDirs.add(licensesDir);
+    }
+
+    public void addLicenseDir(File licenseDir){
+        this.licensesDir(licenseDir);
+    }
+    public List<File> getLicensesDirs(){
+        return Collections.unmodifiableList(this.licensesDirs);
+    }
+
+//    public static String getText(File file){
+//        return getText(file, StandardCharsets.UTF_8);
+//    }
+
+    /**
+     * Add notices from the specified directory.
+     */
+
+    public NoticeTask() {
+        setDescription("Create a notice file from dependencies");
+        // Default licenses directory is ${projectDir}/licenses (if it exists)
+        File licensesDir = new File(getProject().getProjectDir(), "licenses");
+        if (licensesDir.exists()) {
+            licensesDirs.add(licensesDir);
+        }
+
+    }
+
+    @TaskAction
+    public void generateNotice() {
+        final StringBuilder output = new StringBuilder();
+
+        try{
+            Files.lines(this.inputFile.toPath(), StandardCharsets.UTF_8)
+                .forEach(s -> {
+                    output.append(s.trim());
+                    output.append("\n");
+                });
+            output.append("\n\n");
+
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
This is a logging anti pattern as it will most likely cause the same stack trace to be printed twice. 
I would just declare that the method throws IOException